x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/execution/sequence/CircuitBreakerTests.java

```diff
@@ -153,6 +201,283 @@ public void testCircuitBreakerSequnceMatcher() {
+                            emptyList(),
+                            emptySet()
+                        ),
+                        keyExtractors,
+                        tsExtractor,
+                        null,
+                        implicitTbExtractor,
+                        false
+                    )
+                );
+            }
+
+            SequenceMatcher matcher = new SequenceMatcher(sequenceFiltersCount, false, TimeValue.MINUS_ONE, null, eqlCircuitBreaker);
+            TumblingWindow window = new TumblingWindow(eqlClient, criteria, null, matcher);
+            window.execute(wrap(p -> {}, ex -> {}));
+
+            assertTrue(esClient.searchRequestsRemainingCount() == 0); // ensure all the search requests have been asked for
+            assertEquals(0, eqlCircuitBreaker.getTrippedCount()); // the circuit breaker shouldn't trip
+            assertEquals(0, eqlCircuitBreaker.getUsed()); // the circuit breaker memory should be clear
+        }
+    }
+
+    /**
+     *  A type of internal Node client that deals with three types of requests: open PIT, close PIT and SearchRequest.
+     *  This class is used by {@code CircuitBreakerTests#testMemoryClearedOnSuccessfulRequest()} and
+     *  {@code CircuitBreakerTests#testMemoryClearedOnShardsException()} methods to test the circuit breaker memory usage
+     *  in case of a successful sequence request but also for a failed sequence request.
+     */
+    private abstract class ESMockClient extends NoOpClient {
+        private final AtomicLong pitContextCounter = new AtomicLong();
+        protected final CircuitBreaker circuitBreaker;
+        // private final String pitId;
+        private int searchRequestsRemainingCount;
+        private final String pitId = "test_pit_id";
+
+        ESMockClient(CircuitBreaker circuitBreaker, int searchRequestsRemainingCount) {
+            super(getTestName());
+            this.circuitBreaker = circuitBreaker;
+            this.searchRequestsRemainingCount = searchRequestsRemainingCount;
+        }
+
+        abstract <Response extends ActionResponse> void handleSearchRequest(ActionListener<Response> listener, SearchRequest searchRequest);
+
+        @SuppressWarnings("unchecked")
+        @Override
+        protected <Request extends ActionRequest, Response extends ActionResponse> void doExecute(
+            ActionType<Response> action,
+            Request request,
+            ActionListener<Response> listener
+        ) {
+            if (request instanceof OpenPointInTimeRequest) {
+                pitContextCounter.incrementAndGet();
+                OpenPointInTimeResponse response = new OpenPointInTimeResponse(pitId);
+                listener.onResponse((Response) response);
+            } else if (request instanceof ClosePointInTimeRequest) {
+                ClosePointInTimeResponse response = new ClosePointInTimeResponse(true, 1);
+                assert pitContextCounter.get() > 0;
```

Comment:
did you choose the assert (here and below) intentionally?

Reply:
I thought of it as a stronger statement about the pit opening/closing.